app/javascript/consent/ConsentComponent.js

```diff
@@ -24,11 +24,14 @@ type Props = {
   showConsentRequired: boolean,
   // changeConsent: dispatches the change consent action.
   changeConsent: (value: boolean) => void,
+  // consentChanged: callback to the parent component
+  consentChanged: (value: boolean) => void,
 };
 
 class ConsentComponent extends PureComponent<Props> {
   changeConsent = (consented: boolean) => {
     this.props.changeConsent(consented);
+    this.props.consentChanged(consented);
```

Comment:
`changeConsented` here seems unnecessary. Since you need the value in the other component, you can simply fetch it from the store (in `mapStateToProps`, I'll leave a separate comment below).

The second problem with this is that the other components that don't pass it will fail (telling you `this.props.consentChanged is not a function`). This is why Flow is failing as well.